vm/handler.go

```diff
@@ -78,8 +79,13 @@ func (h *DeployTxHandler) ProcessTx(
 		return r, errors.Wrapf(errCreate, "[DeployTxHandler] Error deploying contract on create")
 	}
 
-	reg := h.CreateRegistry(state)
-	reg.Register(tx.Name, addr, caller)
+	if h.AllowNamedEVMContract || tx.VmType == VMType_PLUGIN {
+		reg := h.CreateRegistry(state)
+		err := reg.Register(tx.Name, addr, caller)
+		if err != nil && state.FeatureEnabled(loomchain.CheckRegistryErrorFeature, false) { // if feature enable is on then handle the error
+			return r, err
+		}
+	}
```

Comment:
This should go before `vm, err := h.Manager.InitVM(tx.VmType, state)`:
```golang
version1_1 := state.FeatureEnabled(loomchain.DeployTxVersion1_1Feature, false)
if version1_1 && (tx.VmType == VMType_EVM) && (len(tx.Name) > 0) && !h.AllowNamedEVMContracts {
  return r, errors.New("named evm contracts are not allowed")
}
```

And then the second part can be this:
```golang
if err := reg.Register(tx.Name, addr, caller); err != nil && version1_1 {
  return r, err
}
```